example/src/views/login/screen2.js

```diff
@@ -1,45 +1,416 @@
+      this,
+    )
+    this.signup = this.signup.bind(this)
+  }
+
+  async componentDidMount() {
+    await Font.loadAsync({
+      light: require('../../../assets/fonts/Ubuntu-Light.ttf'),
+      bold: require('../../../assets/fonts/Ubuntu-Bold.ttf'),
+      lightitalic: require('../../../assets/fonts/Ubuntu-Light-Italic.ttf'),
+    })
+
+    this.setState({ fontLoaded: true })
+  }
+
+  signup() {
+    LayoutAnimation.easeInEaseOut()
+    const usernameValid = this.validateUsername()
+    const emailValid = this.validateEmail()
+    const passwordValid = this.validatePassword()
+    const confirmationPasswordValid = this.validateConfirmationPassword()
+    if (
+      emailValid &&
+      passwordValid &&
+      confirmationPasswordValid &&
+      usernameValid
+    ) {
+      this.setState({ isLoading: true })
+      setTimeout(() => {
+        LayoutAnimation.easeInEaseOut()
+        this.setState({ isLoading: false })
+        Alert.alert('🎸', 'You rock')
+      }, 1500)
+    }
+  }
+
+  validateUsername() {
+    const { username } = this.state
+    const usernameValid = username.length > 0
+    LayoutAnimation.easeInEaseOut()
+    this.setState({ usernameValid })
+    usernameValid || this.usernameInput.shake()
+    return usernameValid
+  }
+
+  validateEmail() {
+    const { email } = this.state
+    const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+    const emailValid = re.test(email)
+    LayoutAnimation.easeInEaseOut()
+    this.setState({ emailValid })
+    emailValid || this.emailInput.shake()
+    return emailValid
+  }
+
+  validatePassword() {
+    const { password } = this.state
+    const passwordValid = password.length >= 8
+    LayoutAnimation.easeInEaseOut()
+    this.setState({ passwordValid })
+    passwordValid || this.passwordInput.shake()
+    return passwordValid
+  }
+
+  validateConfirmationPassword() {
+    const { password, confirmationPassword } = this.state
+    const confirmationPasswordValid = password === confirmationPassword
+    LayoutAnimation.easeInEaseOut()
+    this.setState({ confirmationPasswordValid })
+    confirmationPasswordValid || this.confirmationPasswordInput.shake()
+    return confirmationPasswordValid
+  }
+
+  setSelectedType = selectedType =>
+    LayoutAnimation.easeInEaseOut() || this.setState({ selectedType })
 
-export default class LoginScreen2 extends Component {
   render() {
-    return (
-      <View style={styles.container}>
-        <Image
-          source={BG_IMAGE}
-          style={styles.bgImage}
+    const {
+      isLoading,
+      selectedType,
+      fontLoaded,
+      confirmationPassword,
+      email,
+      emailValid,
+      password,
+      passwordValid,
+      confirmationPasswordValid,
+      username,
+      usernameValid,
+    } = this.state
+    return !fontLoaded
+      ? <Text> Loading... </Text>
+			: <ScrollView
+			scrollEnabled={false}
```

Comment:
weird formatting. Also if a return is multi-line consider using () for readability